codec/src/main/java/io/netty/handler/codec/compression/FastLzDecompressor.java

```diff
@@ -208,12 +209,17 @@ public ByteBuf decompress(ByteBuf in, ByteBufAllocator allocator) throws Decompr
 
     @Override
     public boolean isFinished() {
-        return currentState == State.DONE || currentState == State.CORRUPTED;
+        return currentState == State.DONE || currentState == State.CORRUPTED || currentState == State.CLOSED;
+    }
+
+    @Override
+    public boolean isClosed() {
+        return currentState == State.CLOSED;
     }
 
     @Override
     public void close() {
-        currentState = State.DONE;
+         currentState = State.CLOSED;
```

Comment:
Not sure if there was an accidental extra space here, or if it was missing to begin with.